examples/microwave-oven-app/microwave-oven-common/src/microwave-oven-device.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "microwave-oven-device.h"
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::OperationalState;
+using namespace chip::app::Clusters::MicrowaveOvenControl;
+using OperationalStateEnum = chip::app::Clusters::OperationalState::OperationalStateEnum;
+using Status               = Protocols::InteractionModel::Status;
+
+Protocols::InteractionModel::Status
+MicrowaveOvenDevice::HandleMicrowaveOvenSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTime, uint8_t powerSetting)
+{
+    // placeholder implementation
+    // TODO: set Microwave Oven cooking mode by cookMode.Value().
+    mMicrowaveOvenControlInstance.SetCookTime(cookTime);
+    mMicrowaveOvenControlInstance.SetPowerSetting(powerSetting);
+    return Status::Success;
+}
+
+Protocols::InteractionModel::Status MicrowaveOvenDevice::HandleMicrowaveOvenAddMoreTimeCallback(uint32_t addedCookTime)
+{
+    // placeholder implementation
+    mMicrowaveOvenControlInstance.SetCookTime(addedCookTime);
```

Comment:
SetCookTime will replace the cooktime, not add time to the existing time. This will cause this to fail to work as intended. Please update the code to actually *ADD* the cook time to the existing total, rather than replace.

If current cook time left is `35 seconds`, and we call with `addedCookTime == 60` , then cook time left right now with this implementation (before a fix) will be `60 seconds` instead of `35 + 60 == 95 seconds`.

Reply:
The "ADD cook time" logic is implemented in SDK already,  here the variable "addedCookTime" means the original left time plus the "add" time,

seems "addedCookTime" is ambiguous, I will change it to  "finalCookTime".



> SetCookTime will replace the cooktime, not add time to the existing time. This will cause this to fail to work as intended. Please update the code to actually _ADD_ the cook time to the existing total, rather than replace.
> 
> If current cook time left is `35 seconds`, and we call with `addedCookTime == 60` , then cook time left right now with this implementation (before a fix) will be `60 seconds` instead of `35 + 60 == 95 seconds`.